Sprint12 04 a43067 mysql stop start

lopaka opened the pull request:
Stop/start support for mysql master and slave

Commit:
[block_device::setup_ephemeral] Added nick's fix to check both fstab and mtab.
Cosmetic changes - trailing white space, using 'loop' instead of 'while 1', moved variable creation inside 'if' statement using it.

Commit:
Merge branch 'master' into sprint12_04_a43067_mysql_stop_start

Commit:
Merge branch 'master' into sprint12_04_a43067_mysql_stop_start

Commit:
Merge branch 'master' into sprint12_04_a43067_mysql_stop_start

Commit:
[db::install_server] runs db_register_master if this is a master and it is initialized, resetting tags and node attributes.

Commit:
[db::install_server] correctly giving a conditional statement to run db_register_master definition

Commit:
[db::install_server] updating conditional statement to run db_register_master definition

Commit:
[db] removed db_find_master definition and created db_register_slave.
[db::do_init_slave_at_boot] logic added to determine if booting from stop/start
[db::do_primary_init_slave] removed dup code foudn in db_register_slave definition
[db::do_secondary_init_slave] removed dup code foudn in db_register_slave definition
[db::install_server] logic to determine sever coming up from stop/start.
[db::request_master_allow] corrected comments
[db_mysql] providers/default.rb - corrected sys_firewall name texts.  removed duplicate code.
[sys_firewall] providers/default - improved error logs

Commit:
[db_mysql] my.cnf.erb - set relay-logs to be instance_uuid instead of hostname.

Commit:
[db_mysql] helper.rb - defined new method to generate a relay_log name and only create it if it is not already set.
[db_mysql] providers/default.rb - my.cnf.erb has new variable to pass.
[db_mysql] default/my.cnf.erb - using new variable '@relay_log'

Commit:
[db][db_mysql] updated comments

Commit:
Merge branch 'master' into sprint12_04_a43067_mysql_stop_start

douglaswth commented on cookbooks/db/definitions/db_register_slave.rb:
```diff
@@ -12,10 +12,15 @@
 # == Params
+#   :action => :primary_restore - does a primary restore
+#              :secondary_restore - does a secondary restore
+#              :no_restore - will not restore - used for stop/start
```
I've been trying to use YARD syntax in comments like this since we should be able to use our YARD plugin pretty soon (`cookbooks/db/definitions/db_state_set.rb` is one example). Also, you can take a look at the `rightscale_tools` stuff for examples.

douglaswth commented on cookbooks/db_mysql/libraries/helper.rb:
```diff
@@ -19,24 +19,31 @@ module Helper
-            Chef::Log.warn "This database cookbook requires our premium 'rightscale_tools' gem." 
-            Chef::Log.warn "Please contact Rightscale to upgrade your account." 
+            Chef::Log.warn "This database cookbook requires our premium 'rightscale_tools' gem."
+            Chef::Log.warn "Please contact Rightscale to upgrade your account."
```
This messaging should probably be changed since it won't actually be premium.

douglaswth commented:
Other than a few minor cosmetic issues this is ready to merge.

Commit:
[db] definitions/db_register_slave.rb - updated params section of header comment.
[db_mysql] libraries/helper.rb - update logging message by removing 'premium' reference.

Commit:
Merge branch 'master' into sprint12_04_a43067_mysql_stop_start
